base/float.jl

```diff
@@ -850,21 +850,13 @@ truncmask(x, mask) = x
 
 ## Array operations on floating point numbers ##
 
-float(A::AbstractArray{<:AbstractFloat}) = A
-
-function float(A::AbstractArray{T}) where T
-    if !isleaftype(T)
-        error("`float` not defined on abstractly-typed arrays; please convert to a more specific type")
-    end
-    convert(AbstractArray{typeof(float(zero(T)))}, A)
-end
-
-float(r::StepRange) = float(r.start):float(r.step):float(last(r))
-float(r::UnitRange) = float(r.start):float(last(r))
-float(r::StepRangeLen) = StepRangeLen(float(r.ref), float(r.step), length(r), r.offset)
-function float(r::LinSpace)
-    LinSpace(float(r.start), float(r.stop), length(r))
-end
+# Optimizations for broadcasting float()
+broadcast(::typeof(float), A::AbstractArray{<:AbstractFloat}) = A
```

Comment:
Ah, sorry. I missed that "adding a few broadcast specializations" was exactly about this issue. I'm not sure what my opinion is. Always getting a copy is easy to reason about but avoiding an unnecessary copy is also a good thing.